torchtext/experimental/transforms.py

```diff
@@ -0,0 +1,114 @@
+import torch
+import torch.nn as nn
+from typing import List, Tuple
+
+
+__all__ = [
+    'BasicEnglishNormalize',
+    'RegexTokenizer'
+]
+
+
+class BasicEnglishNormalize(nn.Module):
+    r"""Basic normalization for a string sentence.
+
+    Normalization includes
+        - lowercasing
+        - complete some basic text normalization for English words as follows:
+            - add spaces before and after '\''
+            - remove '\"',
+            - add spaces before and after '.'
+            - replace '<br \/>'with single space
+            - add spaces before and after ','
+            - add spaces before and after '('
+            - add spaces before and after ')'
+            - add spaces before and after '!'
+            - add spaces before and after '?'
+            - replace ';' with single space
+            - replace ':' with single space
+            - replace multiple spaces with single space
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import BasicEnglishNormalize
+        >>> test_sample = 'Basic English Normalization for a Line of Text'
+        >>> basic_english_normalize = BasicEnglishNormalize()
+        >>> jit_basic_english_normalize = torch.jit.script(basic_english_normalize)
+        >>> tokens = jit_basic_english_normalize(test_sample)
+    """
+
+    regex_and_replacement_string_pairs: List[Tuple[torch.classes.torchtext.Regex, str]]
+
+    def __init__(self):
+        super(BasicEnglishNormalize, self).__init__()
+        patterns_list = [
+            (r'\'', ' \'  '),
+            (r'\"', ''),
+            (r'\.', ' . '),
+            (r'<br \/>', ' '),
+            (r',', ' , '),
+            (r'\(', ' ( '),
+            (r'\)', ' ) '),
+            (r'\!', ' ! '),
+            (r'\?', ' ? '),
+            (r'\;', ' '),
+            (r'\:', ' '),
+            (r'\s+', ' ')]
+
+        regex_objects = map(lambda pattern_tuple: torch.classes.torchtext.Regex(pattern_tuple[0]), patterns_list)
+        replacement_strings = map(lambda pattern_tuple: pattern_tuple[1], patterns_list)
+        self.regex_and_replacement_string_pairs = list(zip(regex_objects, replacement_strings))
```

Comment:
nit: You don't need to store this imo, you could just do

```
for regex, replacement_string in zip(regex_objects, replacement_strings):
```
below

Reply:
so I wasn't actually storing `regex_objects` or `replacement_strings ` as attributes. since this tokenizer could possibly get called many times for different lines of text, wouldn't it make sense to store `regex_and_replacement_string_pairs` as an attribute so that we don't have to call `zip`? Also avoids the issue of storing 2 different lists when logically the lists are related and should be stored as a list of ordered pairs. LMK what you think

Reply:
There really shouldn't be any performance penalties, but it's a minor thing and indeed a thing you can avoid if you set it up like this.